pkg/clients/tekton/taskruns.go

```diff
@@ -91,6 +93,37 @@ func (t *TektonController) GetTaskRun(name, namespace string) (*pipeline.TaskRun
 	return &taskRun, nil
 }
 
+// StoreTaskRun stores a given TaskRun as an artifact.
+func (t *TektonController) StoreTaskRun(prefix string, taskRun *pipeline.TaskRun) error {
+	artifacts := make(map[string][]byte)
+
+	taskRunYaml, err := yaml.Marshal(taskRun)
+	if err != nil {
+		g.GinkgoWriter.Printf("failed to store taskRun %s:%s: %s\n", taskRun.GetNamespace(), taskRun.GetName(), err.Error())
+	}
+	artifacts["taskRun-"+taskRun.Name+".yaml"] = taskRunYaml
+
+	if err := logs.StoreArtifacts(artifacts); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *TektonController) StoreTaskRunsForPipelineRun(c crclient.Client, pr *pipeline.PipelineRun) error {
+	for _, chr := range pr.Status.ChildReferences {
+		taskRun := &pipeline.TaskRun{}
+		taskRunKey := types.NamespacedName{Namespace: pr.Namespace, Name: chr.Name}
+		if err := c.Get(context.Background(), taskRunKey, taskRun); err != nil {
+			return err
+		}
+		if err := t.StoreTaskRun(taskRun.Name, taskRun); err != nil{
+			g.GinkgoWriter.Printf("failed to store taskRun %s:%s: %s\n", taskRun.GetNamespace(), taskRun.GetName(), err.Error())
```

Comment:
return the error here too, no?

Reply:
It's to print & ignore the error and store more content for debugging. Changed the message which may be misleading